foundations/foundation-metrics/src/main/java/io/servicecomb/foundation/metrics/output/servo/RollingMetricsFileOutput.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2017 Huawei Technologies Co., Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.servicecomb.foundation.metrics.output.servo;
+
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.apache.log4j.PatternLayout;
+import org.apache.log4j.spi.LoggingEvent;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.netflix.servo.publish.BasicMetricFilter;
+import com.netflix.servo.publish.CounterToRateMetricTransform;
+import com.netflix.servo.publish.MetricObserver;
+import com.netflix.servo.publish.MonitorRegistryMetricPoller;
+import com.netflix.servo.publish.PollRunnable;
+import com.netflix.servo.publish.PollScheduler;
+
+import io.servicecomb.foundation.common.utils.RollingFileAppenderExt;
+import io.servicecomb.foundation.metrics.output.MetricsFileOutput;
+import io.servicecomb.serviceregistry.RegistryUtils;
+import io.servicecomb.serviceregistry.api.registry.Microservice;
+
+@Component
+public class RollingMetricsFileOutput extends MetricsFileOutput {
+  private final Map<String, RollingFileAppenderExt> metricsAppenders = new HashMap<>();
+  private final MetricsContentConvertor convertor;
+  private final MetricsContentFormatter formatter;
+  private final String fileNameHeader;
+
+  //auto init when as spring bean
+  @Autowired
+  public RollingMetricsFileOutput(MetricsContentConvertor convertor, MetricsContentFormatter formatter) {
+    this(convertor, formatter, true);
+  }
+
+  public RollingMetricsFileOutput(MetricsContentConvertor convertor, MetricsContentFormatter formatter,
+      boolean autoInit) {
+    if (RegistryUtils.getServiceRegistry() != null) {
+      Microservice microservice = RegistryUtils.getMicroservice();
+      fileNameHeader = String.join(".", microservice.getAppId(), microservice.getServiceName());
+    } else {
+      fileNameHeader = String.join(".", "local", "test");
```

Comment:
Throw the exception here, as we cannot find the microservice  information.

Reply:
Fixed